tests/Unit/DSC_CMForestDiscovery.tests.ps1

```diff
@@ -0,0 +1,259 @@
+                }
+            }
+        }
+        Describe "$moduleResourceName\Set-TargetResource" {
+            Mock -CommandName Import-ConfigMgrPowerShellModule
+            Mock -CommandName Set-Location
+            Mock -CommandName Set-CMDiscoveryMethod
+            Context 'When Set-TargetResource runs successfully' {
+                It 'Should call expected commands enabling discovery' {
+                    Mock -CommandName Get-TargetResource -MockWith { $getReturnDisabled }
+                    Mock -CommandName New-CMSchedule
+                    Set-TargetResource @standardGetInput
+                    Assert-MockCalled Import-ConfigMgrPowerShellModule -Exactly -Times 1 -Scope It
+                    Assert-MockCalled Set-Location -Exactly -Times 2 -Scope It
+                    Assert-MockCalled Get-TargetResource -Exactly -Times 1 -Scope It
+                    Assert-MockCalled New-CMSchedule -Exactly -Times 0 -Scope It
+                    Assert-MockCalled Set-CMDiscoveryMethod -Exactly -Times 1 -Scope It
+                }
+                It 'Should call expected commands enabling discovery and changing the schedule' {
+                    Mock -CommandName Get-TargetResource -MockWith { $getReturnDisabledOutput }
+                    Mock -CommandName New-CMSchedule -MockWith { $scheduleConvertDays } -ParameterFilter { $RecurCount -eq 7 }
+                    Mock -CommandName New-CMSchedule -MockWith { $scheduleConvertDaysMismatch } -ParameterFilter { $RecurCount -eq 6 }
+                    Set-TargetResource @returnEnabledDaysMismatch
+                    Assert-MockCalled Import-ConfigMgrPowerShellModule -Exactly -Times 1 -Scope It
+                    Assert-MockCalled Set-Location -Exactly -Times 2 -Scope It
+                    Assert-MockCalled Get-TargetResource -Exactly -Times 1 -Scope It
+                    Assert-MockCalled New-CMSchedule -Exactly -Times 2 -Scope It
+                    Assert-MockCalled Set-CMDiscoveryMethod -Exactly -Times 1 -Scope It
+                }
+                It 'Should call expected commands disabling discovery' {
+                    Mock -CommandName Get-TargetResource -MockWith { $getReturnEnabledDays }
+                    Mock -CommandName New-CMSchedule
+                    Set-TargetResource @getReturnDisabled
+                    Assert-MockCalled Import-ConfigMgrPowerShellModule -Exactly -Times 1 -Scope It
+                    Assert-MockCalled Set-Location -Exactly -Times 2 -Scope It
+                    Assert-MockCalled Get-TargetResource -Exactly -Times 1 -Scope It
+                    Assert-MockCalled New-CMSchedule -Exactly -Times 0 -Scope It
+                    Assert-MockCalled Set-CMDiscoveryMethod -Exactly -Times 1 -Scope It
+                }
+            }
+            Context 'When running Set-TargetResource should throw' {
+                It 'Should call expected commands and throw if query membership throws' {
```

Comment:
Query Membership?

Reply:
Good catch, this was left over from an example I used to set up throws. Fixed here.